### PR TITLE
Make map and gamemode searchable with searchbar.

### PR DESCRIPTION
**Pull request by laundmo**
This adds the map name and the gamemode name to the serverbrowser search bar, to allow for searching for those.

OLD: ~~This changes the map filter in Better Serverbrowser to a searchbar.~~

### CODE REVIEW

**Comment by abarichello** [comment]
I think the best option for this field would be a dropdown. There's too many maps to cycle through with the current implementation but at the same time adding a searchbar for a limited set of elements also feels wrong

**Comment by laundmo** [comment]
actually, the correct thing would be to include map and gamemodes as part of the normal search. which i have done as a mod, but just haven't gotten around to changing this PR

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 1]
```diff
@@ -684,7 +684,7 @@ void function FilterServerList()
 		{
 			if ( filterArguments.useSearch )
 			{
-				string sName = tempServer.serverName.tolower()
+				string sName = tempServer.serverName.tolower() + " " + GetMapDisplayName(Localize(tempServer.serverMap)).tolower() + " " + tempServer.serverMap.tolower() + " " + tempServer.serverGamemode.tolower()
```
Comment (laundmo) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:687]:
```suggestion
				string sName = tempServer.serverName.tolower() + " " + GetMapDisplayName(Localize(tempServer.serverMap)).tolower() + " " + tempServer.serverMap.tolower() + " " + tempServer.serverGamemode.tolower() + " " + Localize(tempServer.serverGamemode).lower()
```

**Comment by laundmo** [comment]
fixed the fact gamemode name wasnt getting localized

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 2]
```diff
@@ -684,7 +684,7 @@ void function FilterServerList()
 		{
 			if ( filterArguments.useSearch )
 			{
-				string sName = tempServer.serverName.tolower()
+				string sName = tempServer.serverName.tolower() + " " + GetMapDisplayName(Localize(tempServer.serverMap)).tolower() + " " + tempServer.serverMap.tolower() + " " + tempServer.serverGamemode.tolower() + " " + Localize(tempServer.serverGamemode).lower()
```
Comment (laundmo) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:687]:
```suggestion
				string sName = tempServer.serverName.tolower() + " " + Localize(GetMapDisplayName(tempServer.serverMap)).tolower() + " " + tempServer.serverMap.tolower() + " " + tempServer.serverGamemode.tolower() + " " + Localize(tempServer.serverGamemode).tolower()

```

**Comment by laundmo** [comment]
ok NOW its fixed

**Pull request merged**